packages/sdk/src/utils.ts

```diff
@@ -110,3 +136,11 @@ export const COMBINED_STATUS_WEIGHTS: Record<RequestStatus | TimeoutStatus, numb
 	[TimeoutStatus.HYPERBRIDGE_FINALIZED_TIMEOUT]: 8,
 	[TimeoutStatus.TIMED_OUT]: 9,
 }
+
+export async function latestStateMachineHeight(wsUrl: string, stateMachineId: StateMachineIdParams): Promise<bigint> {
```

Comment:
can we reuse the api object in the substrate client here? You might need to add this method to the `IChain` interface.

Reply:
The problem is this creates a new connection which js isn't great at deallocating resoruces, So best to reuse any existing connections to prevent memory leaks

Reply:
I think this makes sense, but the clients might be for the different hyperbridge networks. It may make sense to  have the indexer client instance hold the substrate clients and expose methods for clean up. But yes, this can be done in a seperate PR